http.go

```diff
@@ -315,20 +315,29 @@ func (resp *Response) LocalAddr() net.Addr {
 	return resp.laddr
 }
 
-// Body returns response body.
-//
-// The returned body is valid until the response modification.
-func (resp *Response) Body() []byte {
+// BodyOrErr returns response (body, nil) if success.
+// when error occurs, it returns nil, err.
+func (resp *Response) BodyOrErr() ([]byte, error) {
 	if resp.bodyStream != nil {
 		bodyBuf := resp.bodyBuffer()
 		bodyBuf.Reset()
 		_, err := copyZeroAlloc(bodyBuf, resp.bodyStream)
 		resp.closeBodyStream() //nolint:errcheck
 		if err != nil {
-			bodyBuf.SetString(err.Error())
+			return nil, err
 		}
 	}
-	return resp.bodyBytes()
+	return resp.bodyBytes(), nil
+}
+
+// Body returns response body.
+// if error occurs, the error message would be put into body.
+func (resp *Response) Body() []byte {
+	body, err := resp.BodyOrErr()
+	if err != nil {
+		return []byte(err.Error())
```

Comment:
I'm afraid that this is not the same as `bodyBuf.SetString(err.Error())`. `bodyBuf.SetString` also modifies the `Response` body for if it gets used afterwards. I think the implementation of `Body()` should just stay the same as before.